### PR TITLE
new py-ultralytics package

### PR DESCRIPTION
**Pull request by Sangu-Mbekelu**
depends on #35889

### CODE REVIEW

**Comment by Sangu-Mbekelu** [comment]
@spackbot fix style

**Review by tldahlgren — COMMENTED** [review]
Confirmed the sha256.

File: var/spack/repos/builtin/packages/py-ultralytics/package.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    depends_on("py-requests@2.23.0:", type=("build", "run"))
+    depends_on("py-scipy@1.4.1:", type=("build", "run"))
+    depends_on("py-torch@1.7.0:", type=("build", "run"))
+    depends_on("py-torchvision@0.8.1:", type=("build", "run"))
+    depends_on("py-tqdm@4.64.0:", type=("build", "run"))
+    depends_on("py-tensorboard@2.4.1:", type=("build", "run"))
+    depends_on("py-pandas@1.1.4:", type=("build", "run"))
+    depends_on("py-seaborn@0.11.0:", type=("build", "run"))
+    depends_on("py-psutil", type=("build", "run"))
+    depends_on("py-thop@0.1.1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ultralytics/package.py:36]: Missing deps on `py-certifi@2022.12.7:` and `py-sentry-sdk`

File: var/spack/repos/builtin/packages/py-ultralytics/package.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    homepage = "https://github.com/ultralytics/ultralytics"
+    pypi = "ultralytics/ultralytics-8.0.50.tar.gz"
+
+    version("8.0.50", sha256="fdcb22300b63b72aa52da1713c33c01741aca031a61f15327eb6f02615bb4b97")
+
+    depends_on("py-setuptools", type=("build", "run"))
+    depends_on("py-matplotlib@3.2.2:", type=("build", "run"))
+    depends_on("py-numpy@1.18.5:", type=("build", "run"))
+    depends_on("opencv@4.6.0:", type=("build", "run"))
+    depends_on("py-pillow@7.1.2:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ultralytics/package.py]:
```suggestion
    depends_on("pil@7.1.2:", type=("build", "run"))
```
We have a virtual provider for this

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — DISMISSED** [review]

**Review by bernhardkaindl — DISMISSED** [review]
Rebased onto the new develop branch to have `py-certifi@2022.12.7` which was just merged.

File: var/spack/repos/builtin/packages/py-ultralytics/package.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    homepage = "https://github.com/ultralytics/ultralytics"
+    pypi = "ultralytics/ultralytics-8.0.50.tar.gz"
+
+    version("8.0.50", sha256="fdcb22300b63b72aa52da1713c33c01741aca031a61f15327eb6f02615bb4b97")
+
+    depends_on("py-setuptools", type=("build", "run"))
+    depends_on("py-matplotlib@3.2.2:", type=("build", "run"))
+    depends_on("py-numpy@1.18.5:", type=("build", "run"))
+    depends_on("opencv@4.6.0:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ultralytics/package.py]:
```suggestion
    depends_on("opencv@4.6.0:+python3", type=("build", "run"))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**